lidur1.md

```diff
@@ -0,0 +1,183 @@
+# *Spurning 1: Leita af kennitölu:*
+
+    I. Skrifið tvær reglulegar segðir til að finna kennitölur einstaklinga og   fyrirtækja í texta.
+
+    II. Skrifið Python kóða sem leitar að kennitölum samkvæmt reglulegu segðunum og prentar niðurstöðurnar.
+    
+    III. Keyrið kóðann með valmöguleikum til að leita annað hvort af einstaklinga, fyrirtækja eða báðum í einu.
+
+## *LAUSN:*
+Byrjum á að svara fyrstu spurningunni þar sem við eigum að skrifa tvær reglulegar segðir til að finna kennitölur einstaklinga og á hinn bóginn kennitölur fyrirtækja í texta:
+
+**Forsendur: Samkvæmt Þjóðskrá Íslands** 
+Form einstaklings kennitalna á Íslandi er alltaf það sama þ.e. Kennitalann er í heildina 10 tölustafir þar sem fyrstu 6 eru myndaðir af fæðingardagsetningu einstaklingsins.  
+Næstu tveir tölustafir hafa í raun enga merkingu og er þeim alla jafna úthlutað í röð frá og með 20, t.d. 120160-33.   
+Því næst er svokölluð öryggistalan reiknuð út en hún má vera á bilinu 0 til 9. Aftasti stafur í kennitölu er 8, 9 eða 0 og merkir hann öldina sem viðkomandi er fæddur á. Kt. gæti verið t.d. 129160-3359
+
+Form fyrirtækis kennitalna á Íslandi er þó aðeins frábrugðið. Kennitalan er í heild sinni 10 tölustafir myndaðir af stofnunardagsetningu fyrirtækisins en fyrsta talan er hækkuð um 4. Sem dæmi segjum að fyrirtæki sé stofnað 05.11.2005 þá yrði fyrstu 6 tölustafirnir án upphækkunarinnar 051105 en á réttu formi yrði fyrsta talan hækkuð um 4 og því yrði kt. 451105. Því getur fyrsta talann einungis verið 4, 5, 6 eða 7. Vegna 0+4, 1+4, 2+4, 3+4. Enginn mánuður hefur 40 daga. Síðustu 4 virka þannig að tölustafirnir 7. og 8. geta verið hvaða tala frá 01-99, 9. talan getur verið hvaða tala frá 0-9 og 10. talann er ein af þremur tölum 8,9 eða 0. Kt. gæti verið t.d. 660487-3459
+
+Því getum við byrjað að skrifa reglulega segð sem finnur kennitölu einstaklings og svo fyrirtækis.  
+
+**Einstaklings regluleg segð gæti litið svona út:**
+```regex
+\b(?:0[1-9]|[1-2][0-9]|3[0-1])(?:0[1-9]|1[0-2])\d{2}-(?:2[0-9]|[3-9]\d)\d[890]\b
+```
+
+Þar sem, 
+```regex
+\b: Segða eða orðaamörk
+```
+Þetta merkir „mörk“ segðar og tryggir að leitun byrji á stöðum þar sem orð endar eða byrjar. 
+
+```regex
+(0[1-9]|[1-2][0-9]|3[0-1]): Dagur í Mánuði
+```
+Hér er notað hópur með „valkostum“ (|), sem merkir að það passar eitt af eftirfarandi:  
+**Valkostur 1** - 0[1-9]: Passar að á eftir 0 fylgir tölustafur frá 1 til 9, sem gefur til kynna dagana 01 til 09.  
```

Comment:
Þessar línur er mjög oft endurtekna (að neðan), eða línur sem eru næstum því eins, gerir þetta að wall of text, kannski hægt að stita textan .